src/ripple/rpc/handlers/WalletPropose.cpp

```diff
@@ -31,9 +31,30 @@
 #include <ripple/rpc/handlers/WalletPropose.h>
 #include <ed25519-donna/ed25519.h>
 #include <boost/optional.hpp>
+#include <cmath>
+#include <map>
 
 namespace ripple {
 
+double
+shannon_entropy (std::string const& input)
+{
+    std::map<int, double> freq;
+
+    for (auto const& c : input)
+        freq[c]++;
+
+    double se = 0;
+
+    for (auto& f : freq)
+        f.second /= input.length();
+
+    for (auto const& f : freq)
+        se += (f.second) * log2(f.second);
+
+    return std::floor (-se * input.length());
```

Comment:
Here I'll ask for my own education.  This code review led me to read up on Shannon Entropy (yes, my education is sadly lacking).  The stuff I looked at does not suggest that the entropy takes on discrete values.  So I'm curious why you take the `floor` of the calculated entropy.